src/Moq/FluentMockVisitor.cs

```diff
@@ -173,11 +178,10 @@ private MethodInfo GetTargetMethod(Type objectType, Type returnType)
 				if (Mock.TryGetFromReturnValue(returnValue, out innerMock))
 				{
 					Mock.SetupAllProperties(innerMock);
-					mock.AddFixedReturnValueSetup(info, returnValue);
+					mock.AddFixedReturnValueSetup(info, arguments, setup, returnValue);
 				}
 			}
 
-			mock.Setup(setup).Returns((TResult)returnValue);
```

Comment:
Originally, this setup partially made up for missing argument information in the `InnerMocks` record. It might be an error to move this into a conditional code block when previously, the setup was always recorded. Perhaps we need a unit test to see what difference this could possibly make.